packages/grid/_modules_/grid/models/api/coreApi.ts

```diff
@@ -25,4 +25,9 @@ export interface CoreApi extends EventEmitter {
    * Display the error overlay component.
    */
   showError: (props: any) => void;
+
```

Comment:
> I'm not hands-on in the grid, so it's only feedback from a high-level perspective, the change seems to go in the opposite direction I was expecting. We are introducing a leaking side effect, an update scheduled in one use ref call can be canceled and never called again in another update, this will likely create bugs. I mean, it only works if the use raf hooks are used in the same component. Have you considered to share the update function instead of the then timer? This would guarantee the constraint and hence avoid dropped updates.
> 
> I don't understand why we need to use Raf in the first place. react-virtualized, react-window, react-virtuoso, react-virtual barely uses it. I would encourage we remove this hook from the codebase to make usage of Raf painful, the default solution being to defer rendering prioritization to React.

can be canceled and never called again in another update => nope it is only cancelled when there is another one coming.

Reply:
> can be canceled and never called again in another update => nope it is only cancelled when there is another one coming.

Exactly so it can create bugs. Imagine the following dom structure, where both component use the hook

Component A
  Component B

If component B cancel the raf, component A is never rendered.

Reply:
That's not how it works.
Those are dummy updates. The state is always updated outside of the raf. The raf is just a way to trigger an update of the UI with the latest state at any given time

Reply:
> The raf is ~just~ a way to trigger an update of the UI with the latest state at any given time

Yes, this is exactly my concern, the UI won't reflect the state => bug

Reply:
Of course the UI reflects the state, because it renders the state that we have updated.

Reply:
Not in the above case: https://github.com/mui-org/material-ui-x/pull/506#discussion_r513366096

Reply:
try!

Reply:
If component B cancels the RAF that was scheduled by component A => All components will rerender it doesn't matter who triggers the update. Then react will calculate which components really need to render.

Reply:
> All components will rerender

Why would component A rerender in this case? It's scheduled update was canned, and component B rerendering, only triggers its children, not its parent.

Reply:
Rendering works with props passed between components.
There is no scheduled update for any component. The grid state is central, so triggering a rafUpdate will rerender the gridComponent, however as we have optimized all the children then only those who changed will update.

Reply:
Ok, so we are on the same page (`it only works if the use raf hooks are used in the same component.`), this only works because we don't have:

- Component A
 - Component B

But instead:

- Component A
  - hook 1
  - hook 2
  - hook 3

Hence my initial suggestion: 

> Have you considered to share the update function instead of the then timer? This would guarantee the constraint and hence avoid dropped updates.

Reply:
Hold on, let me try dropping the RAF and see how it goes with the performance.

Reply:
There are no dropped updates and we have many nested components.

Why would we do that? How is that different from what we do?
Why wouldn't it work?

It works with the context so if the context change then it updates the whole react tree. So if a nested components uses setGridState it will work!

Try.